components/brave_ads/core/internal/history/history_item_util_unittest.cc

```diff
@@ -36,6 +36,7 @@ TEST_F(BraveAdsHistoryItemUtilTest, BuildHistoryItem) {
   expected_history_item.ad_content.creative_set_id = ad.creative_set_id;
   expected_history_item.ad_content.campaign_id = ad.campaign_id;
   expected_history_item.ad_content.advertiser_id = ad.advertiser_id;
+  expected_history_item.ad_content.advertiser_id = ad.segment;
```

Comment:
This change highlighted a bug in `ad_content_info.cc` where we were not comparing segments